pinc/ProjectTransition.inc

```diff
@@ -163,6 +164,14 @@ class ProjectTransition
         die("transition has bad 'restriction' value: '$simple_who_restriction'");
     }
 
+    function is_disabled($project)
```

Comment:
I was considering future proofing it, but I was not sure if that would be over complicating things.

In cacb48f2f51826f7bcb5c24f83183801f21b0704, I ended up going for the approach of having `why_disabled` which returns the reason or an empty string if there is no reason and `is_disabled` which simply checks the `why_disabled` function. `is_disabled` checks `why_disabled` to avoid code duplication and makes it easier to maintain since we don't have to maintain conditions in two places.

Arguably we don't really need `is_disabled` any more since we can just check `why_disabled` manually but since they're both pretty simple and fast functions, we could leave them both as a convenience.